### PR TITLE
[Misc][MiniCPM-o] Let a deploy config set Token2Wav's guidance weight

### PR DESCRIPTION
**Pull request by Junbo2002**
## Purpose

> Team: **5.6-sol**

A MiniCPM-o 4.5 Code2Wav stage can set `token2wav_n_timesteps` from its `extra`
config. It cannot set `inference_cfg_rate` at all — and that is the one Token2Wav
constant that has to move *with* the step count.

Classifier-free guidance is applied **once per solver step**, so the weight that
suits one schedule is not the weight that suits another. There are four open PRs
on this branch changing `token2wav_n_timesteps` right now (#6465, #6386, #6589,
#6616) and none of them can sweep the pair without patching the tree, because the
value only exists inside the checkpoint.

This adds `token2wav_cfg_rate` next to it:

```yaml
extra:
  token2wav_n_timesteps: 3
  token2wav_cfg_rate: 0.3
```

**The default is unset**, so nothing changes for anyone who does not ask for it —
the checkpoint's own value is used exactly as before. There is no runtime cost
either way: CFG is `torch.cat((x, x), dim=0)`, which widens each kernel rather
than adding any, so only the weight moves.

`apply_token2wav_cfg_rate` **raises** if the flow decoder exposes no
`inference_cfg_rate`, rather than warning and carrying on. A configuration key
that silently does nothing is worse than a startup error, and this one would be
silent in exactly the situation where somebody is trying to measure its effect.

The measured reason this matters — a sweep on 300 Seed-TTS zh rows where 0.3
improves UTMOS, CER and SIM simultaneously at one flow step — is in #6907. This PR deliberately carries **no default change**; picking a
value is a per-schedule decision and belongs with whoever picks the schedule.

## Test Plan

**vLLM Version:** `0.1.dev1+ge5588e49b` (built from `vllm-project/vllm@e5588e49b`)

**vLLM-Omni Commit:** `ecd9d99da`

`tests/model_executor/models/minicpmo_4_5/test_code2wav_batching.py`, 4 new
pure-CPU cases: the override reaches `flow.decoder.inference_cfg_rate` and
accepts a string; and it raises rather than passing silently for each of the
three ways the decoder can be missing.

## Test Result

`tests/model_executor/models/minicpmo_4_5/`, same machine, same interpreter:

| tree | result |
| --- | --- |
| `ecd9d99da` | 91 passed, 3 skipped, 13 errors |
| this branch | **95 passed**, 3 skipped, 13 errors |

The 13 errors are on the base too — this box has no `pytest-mock`.

### CODE REVIEW

**Comment by Junbo2002** [comment]
Self-review, as `vllm-omni-review-bot` asks for before reviewers are assigned.

**Blast radius.** One `extra` key that defaults to unset, one module-level
helper, and four tests. If `token2wav_cfg_rate` is absent — which it is
everywhere today, including in the deploy configs in this repo — the new code
path is one `dict.get` returning `None` and nothing else runs. The checkpoint's
own `inference_cfg_rate` is used exactly as before.

**The two things I would attack first if this landed in my queue:**

1. *Why raise instead of warn?* Because the failure this key can have is
   invisible. Somebody sets it to sweep a value, the decoder does not have the
   attribute, and they get a table of identical numbers and conclude the
   parameter does not matter. That is a worse outcome than a startup error, and
   it is not hypothetical — the first sweep I ran on this parameter produced four
   bit-identical arms, for a neighbouring reason (a key added to a deploy YAML
   that the harness then replaced with its own file). A key that is set and
   cannot be honoured should say so.

2. *Should this ship a new default?* No, and deliberately not. The right value
   depends on the solver schedule, which is what #6465, #6386, #6589 and #6616
   are each changing to something different. Picking it belongs with whoever
   picks the schedule. What I can contribute is the coupling and a measured
   sweep, which is #6907, and the ability to set the thing, which is this PR.
   Splitting them that way also keeps this diff reviewable on its own merits.

**Reachability.** The value is applied to `token2wav.flow.decoder` immediately
after `Token2wav(...)` returns and before `BatchedToken2Wav` wraps it, which is
the same point `token2wav_n_timesteps` takes effect, so the two behave the same
way with respect to config precedence and logging.

Team: **5.6-sol**.

**Comment by Junbo2002** [comment]
Credit where it is due, and a correction to my own description.

While reviewing what else touches this knob I found that **#5828**
(@OrangePure, closed 2026-08-07) already carried a `token2wav_cfg_rate`
override read out of `extra`, i.e. the same knob this PR adds. I did not know
that when I opened this, and the description should have pointed at it.

The two are not identical — this PR applies the value through a named function,
raises when the decoder exposes no `inference_cfg_rate` instead of warning past
it, and ships tests; and the motivation here is the recalibration argument in
#6907 (the shipped `inference_cfg_rate` was tuned for a flow-step count the
defaults no longer use) rather than CFG-free decoding. But #5828 was first on
the configuration knob itself, and if the maintainers would rather revive that
PR than take this one, I have no objection.
